### ISSUE TITLE
ERC20 Precision API for Clients

### ISSUE THREAD

**Issue by dimalinux** [AthanorLabs/atomic-swap#275]
We need an API call to get the token name and decimals using the contract address for clients like `swapcli`, which need that info to display the token information correctly. As part of the work, we should review the code to make sure we do proper validation on the number of decimals throughout the flow.  For example, we should reject provides amounts that use too many decimal places. We also should look into rounding in the proper direction when computing min/max values for the user in the provided token.

**Comment by doonte** [comment]
Usually when it comes to token math, decimals are only used for displaying in the front end while big integers are used to track the value. I wonder if a switch from decimals to big integers for storing units of value would be useful. After all, the `decimals()` function is optional in the [ERC-20 standard](https://eips.ethereum.org/EIPS/eip-20).

**Comment by dimalinux** [comment]
@doonte  I think what you are requesting was implemented a couple weeks ago in this PR: https://github.com/AthanorLabs/atomic-swap/pull/269. We removed all references to `big.Float` and `float64` from the code base and now use an arbitrary precision decimal library: https://github.com/cockroachdb/apd. This library stores values and does math with big integers, and tracks the decimal point separately in a 2nd field.

I'm looking at the ERC-20 standard you linked to. If the `decimals()` function is not implemented, we can still treat those tokens the same as tokens that do implement it. For our purposes, the value would just be zero.
